client/src/cmdlfem4x50.c

```diff
@@ -0,0 +1,720 @@
+// Copyright (C) 2020 tharexde
+//
+// This code is licensed to you under the terms of the GNU GPL, version 2 or,
+// at your option, any later version. See the LICENSE.txt file for the text of
+// the license.
+//-----------------------------------------------------------------------------
+// Low frequency EM4x50 commands
+//-----------------------------------------------------------------------------
+
+#include "cmdlfem4x50.h"
+#include <ctype.h>
+#include "fileutils.h"
+#include "comms.h"
+#include "em4x50.h"
+
+#define EM4x50_FCT_STDREAD  0
+#define EM4x50_FCT_LOGIN    1
+#define EM4x50_FCT_WRITE    2
+
+int usage_lf_em4x50_info(void) {
+    PrintAndLogEx(NORMAL, "Read all information of EM4x50. Tag nust be on antenna.");
+    PrintAndLogEx(NORMAL, "");
+    PrintAndLogEx(NORMAL, "Usage:  lf em 4x50_info [h] [v] [p <pwd>]");
+    PrintAndLogEx(NORMAL, "Options:");
+    PrintAndLogEx(NORMAL, "       h         - this help");
+    PrintAndLogEx(NORMAL, "       v         - verbose output");
+    PrintAndLogEx(NORMAL, "       p <pwd>   - password (hex) (optional)");
+    PrintAndLogEx(NORMAL, "Examples:");
+    PrintAndLogEx(NORMAL, "      lf em 4x50_info");
+    PrintAndLogEx(NORMAL, "      lf em 4x50_info p fa225de1\n");
+    PrintAndLogEx(NORMAL, "      lf em 4x50_info v p fa225de1\n");
+    return PM3_SUCCESS;
+}
+int usage_lf_em4x50_write(void) {
+    PrintAndLogEx(NORMAL, "Write EM4x50 word. Tag must be on antenna. ");
+    PrintAndLogEx(NORMAL, "");
+    PrintAndLogEx(NORMAL, "Usage:  lf em 4x50_write [h] a <address> w <data>");
+    PrintAndLogEx(NORMAL, "Options:");
+    PrintAndLogEx(NORMAL, "       h         - this help");
+    PrintAndLogEx(NORMAL, "       a <addr>  - memory address to write to (dec)");
+    PrintAndLogEx(NORMAL, "       w <word>  - word to write (hex)");
+    PrintAndLogEx(NORMAL, "       p <pwd>   - password (hex) (optional)");
+    PrintAndLogEx(NORMAL, "Examples:");
+    PrintAndLogEx(NORMAL, "      lf em 4x50_write a 3 w deadc0de");
+    return PM3_SUCCESS;
+}
+int usage_lf_em4x50_write_password(void) {
+    PrintAndLogEx(NORMAL, "Write EM4x50 password. Tag must be on antenna. ");
+    PrintAndLogEx(NORMAL, "");
+    PrintAndLogEx(NORMAL, "Usage:  lf em 4x50_write_password [h] p <pwd> n <pwd>");
+    PrintAndLogEx(NORMAL, "Options:");
+    PrintAndLogEx(NORMAL, "       h         - this help");
+    PrintAndLogEx(NORMAL, "       p <pwd>   - password (hex)");
+    PrintAndLogEx(NORMAL, "       n <pwd>   - new password (hex)");
+    PrintAndLogEx(NORMAL, "Examples:");
+    PrintAndLogEx(NORMAL, "      lf em 4x50_write_password p 11223344 n 01020304");
+    return PM3_SUCCESS;
+}
+
+static uint8_t msb2lsb(uint8_t byte) {
+
+    // convert given byte (msb) into lsb format
+    
+    uint8_t tmp = 0;
+    
+    for (int i = 0; i < 8; i++)
+        tmp |= ((byte >> (7-i)) & 1) << i;
+    
+    return tmp;
+}
+
+static bool check_bit_in_byte(uint8_t pos, uint8_t byte) {
```

Comment:
Would need to unify these check bits functions...
https://github.com/RfidResearchGroup/proxmark3/blob/master/client/src/cmdhficlass.c#L379-L385

Reply:
function "check_bit_in_byte(...)" didn't really make sense - deleted